src/internal/ExpandTransitionChild.js

```diff
@@ -31,39 +36,46 @@ class ExpandTransitionChild extends Component {
     callback();
   }
 
+  componentDidAppear() {
+    this.setAutoHeight();
+  }
+
   componentWillEnter(callback) {
-    const {enterDelay} = this.props;
+    const {enterDelay, transitionDelay, transitionDuration} = this.props;
     const {style} = ReactDOM.findDOMNode(this);
     style.height = 0;
-
-    if (enterDelay) {
-      this.enterTimer = setTimeout(() => callback(), 450);
-      return;
-    }
-
-    callback();
+    this.enterTimer = setTimeout(() => this.open(), enterDelay);
+    this.enteredTimer = setTimeout(() => callback(), enterDelay + transitionDelay + transitionDuration);
   }
 
   componentDidEnter() {
-    this.open();
+    this.setAutoHeight();
   }
 
   componentWillLeave(callback) {
-    const style = ReactDOM.findDOMNode(this).style;
-    style.height = this.refs.wrapper.clientHeight;
+    const {transitionDelay, transitionDuration} = this.props;
+    const {style} = ReactDOM.findDOMNode(this);
+    style.height = `${this.refs.wrapper.clientHeight}px`;
```

Comment:
@oliviertassinari I just force pushed an update to put this back. Thanks again @eng1neer !!